java/code/src/com/suse/manager/webui/services/SaltServerActionService.java

```diff
@@ -1261,10 +1261,21 @@ private Map<LocalCall<?>, List<MinionSummary>> hardwareRefreshListAction(
     }
 
     private Map<LocalCall<?>, List<MinionSummary>> rebootAction(List<MinionSummary> minionSummaries) {
-        Map<LocalCall<?>, List<MinionSummary>> ret = new HashMap<>();
-        ret.put(com.suse.salt.netapi.calls.modules.System
-                .reboot(Optional.of(3)), minionSummaries);
-        return ret;
+        return minionSummaries.stream().collect(
+            Collectors.groupingBy(
+                m -> m.isTransactionalUpdate() ? reboot() :
+                        com.suse.salt.netapi.calls.modules.System.reboot(Optional.of(3))
+            )
+        );
+    }
+
+    /**
+     * This method is temporarily here until a new version of salt-netapi-client that contains it is released
+     */
+    private static LocalCall<String> reboot() {
```

Comment:
I would add a preemptive `@Deprecated` annotation with some description here so it would remind us to remove it when the library is updated.